lightning/src/offers/merkle.rs

```diff
@@ -249,6 +249,7 @@ impl<'a> TlvStream<'a> {
 }
 
 /// A slice into a [`TlvStream`] for a record.
+#[derive(Eq, PartialEq)]
```

Comment:
We only need to compare `record_bytes`, so we can have `from_same_offer` map to that before comparing in order to avoid redundant checks.